lib/experimental/class-gutenberg-rest-comment-controller.php

```diff
@@ -516,6 +527,46 @@ public function get_item_schema() {
 
 		return $schema;
 	}
+
+	/**
+	 * If empty comments are not allowed, checks if the provided comment content is not empty.
+	 *
+	 * @since 6.9.0
+	 *
+	 * @param array $prepared_comment The prepared comment data.
+	 * @return bool True if the content is allowed, false otherwise.
+	 */
+	protected function check_is_comment_content_allowed( $prepared_comment ) {
+		$check = wp_parse_args(
+			$prepared_comment,
+			array(
+				'comment_post_ID'      => 0,
+				'comment_author'       => null,
+				'comment_author_email' => null,
+				'comment_author_url'   => null,
+				'comment_parent'       => 0,
+				'user_id'              => 0,
+			)
+		);
+
+		/** This filter is documented in wp-includes/comment.php */
+		$allow_empty = apply_filters( 'allow_empty_comment', false, $check );
+
+		if ( $allow_empty ) {
+			return true;
+		}
+
+		// Allow empty block comments with resolution metadata [backport].
+		if ( isset( $check['meta']['_wp_block_comment_status'] ) ) {
+			return true;
+		}
```

Comment:
```suggestion
		// Allow empty block comments with resolution metadata [backport].
		if (
			isset( $check['comment_type'] ) &&
			'block_comment' === $check['comment_type'] &&
			isset( $check['meta']['_wp_block_comment_status'] )
		) {
			return true;
		}
```

This conditional statement allows empty comments even if they are not Notes (Block comment), as long as they have the `_wp_block_comment_status` meta.

Reply:
For some reason at this point in code the `comment_type` is having value as `comment`
Although in DB they are getting saved correctly with `block_comment` type

Reverting temporarily till I find the cause of this...

Reply:
@yashjawale I think this is because you're using old request parameters on the JS side. See https://github.com/WordPress/gutenberg/pull/72205/files for more details.

The code below should work correctly.

```diff
diff --git a/lib/experimental/class-gutenberg-rest-comment-controller.php b/lib/experimental/class-gutenberg-rest-comment-controller.php
index ba1a72efe7..8f19c91b02 100644
--- a/lib/experimental/class-gutenberg-rest-comment-controller.php
+++ b/lib/experimental/class-gutenberg-rest-comment-controller.php
@@ -395,12 +395,6 @@ class Gutenberg_REST_Comment_Controller extends WP_REST_Comments_Controller {
                        return $prepared_comment['comment_approved'];
                }
 
-               // [backport].
-               if ( ! empty( $request['comment_type'] ) && 'block_comment' === $request['comment_type'] ) {
-                       $prepared_comment['comment_type']     = $request['comment_type'];
-                       $prepared_comment['comment_approved'] = $request['comment_approved'];
-               }
-
                /**
                 * Filters a comment before it is inserted via the REST API.
                 *
@@ -557,7 +551,11 @@ class Gutenberg_REST_Comment_Controller extends WP_REST_Comments_Controller {
                }
 
                // Allow empty block comments with resolution metadata [backport].
-               if ( isset( $check['meta']['_wp_block_comment_status'] ) ) {
+               if (
+                       isset( $check['comment_type'] ) &&
+                       'block_comment' === $check['comment_type'] &&
+                       isset( $check['meta']['_wp_block_comment_status'] )
+               ) {
                        return true;
                }
 
diff --git a/packages/editor/src/components/collab-sidebar/hooks.js b/packages/editor/src/components/collab-sidebar/hooks.js
index 63a6860f8b..826491580e 100644
--- a/packages/editor/src/components/collab-sidebar/hooks.js
+++ b/packages/editor/src/components/collab-sidebar/hooks.js
@@ -247,8 +247,8 @@ export function useBlockCommentsActions( reflowComments ) {
                                const newCommentData = {
                                        post: getCurrentPostId(),
                                        content: content || '', // Empty content for resolve, content for reopen.
-                                       comment_type: 'block_comment',
-                                       comment_approved: status === 'approved' ? 1 : 0,
+                                       type: 'block_comment',
+                                       status,
                                        parent: id,
                                        meta: {
                                                _wp_block_comment_status:
```

Reply:
Oh I see... Thanks!